### DISCUSSION TITLE
```ls``` without buffering is a nearly transparent algorithm.

### DISCUSSION THREAD

**Discussion by Whammo** [jkotlinski/durexforth#550] [Show and tell]
```forth
require io

: ls
parse-name ?dup if
else drop s" $"
then
8 0 open ioabort
clrchn
8 chkin ioabort
chrin chrin 2drop page
chrin drop
begin chrin drop more
chrin chrin 8 lshift + .
begin chrin ?dup while emit repeat
cr chrin 0= until
clrchn 8 close
;
```

**Comment by jkotlinski** [comment]
Nice! It is clearly more user-friendly, as the screen output starts sooner.

**Comment by ekipan** [comment]
An edge case: I just happened to have exactly 21 files on my disk and the interpreter's ok prompt scrolled the directory heading off the top of the screen. The built-in `ls` didn't have an empty line so the ok prompt didn't do that. I put a 22nd file and the built-in `ls` scrolled off the header, whereas your `ls` did `more`. 

Maybe not an edge case worth caring about since `ls` is just a convenience.

**Reply by ekipan** [reply]
I think you can match the behavior by ~moving `cr more` before any output~ deleting the `cr` at the end. Might also be a good idea to put `more` at the beginning of the loop anyway. Maybe it doesn't matter.

**Comment by Whammo** [comment]
Getting a good look at the directory, #324

 I'm looking for balance.  😄
